### ISSUE TITLE
Node query yields hidden nodes when desktop set to monocle

### ISSUE THREAD

**Issue by eigengrau** [baskerville/bspwm#650]
I’ve noticed (on HEAD) that using something like `bspc query -N -n east` or `bspc node -f east` will return nodes on the current desktop which are not shown due to monocle. This used to (iirc) return error instead, so one could do `bspc node -f east || monitor -f east`. Now, `bspc node -f east` on a monocle workspace will focus a hidden node and hide the node that is currently focused.

Is this by design or in error? I’ve also noticed that `bspc node -f east` will cycle to a hidden node that isn’t actually located east when disengaging monocle.

**Comment by eigengrau** [comment]
Note that this only triggers when the monocle workspace doesn’t also have another monitor in the same direction. So, when I’m on my rightmost monitor, with the workspace in monocle, `bspc node -f west` will not focus a hidden node. But `bspc node -f east`, instead of returning error, will focus a hidden node.

**Comment by baskerville** [comment]
Thanks.

**Comment by aminb** [comment]
This change has turned out to be pretty annoying for me. Is there a way to fix the original issue through some other mean?

Right now with the 7561be4 commit, if I have two floating windows of the same size underneath each other (e.g. two small terminals) I can't switch between them with `super + {j,k}` anymore. Also, if I have one large window and one small window on top of it, once I focus the larger window I can't give the focus back to the smaller window using `super + {j,k}` anymore.

**Comment by baskerville** [comment]
Maybe we can agree on a weaker constraint: if `A` and `B` are identical rectangles then the proposition `A is on the DIR side of B` is false for every `DIR` in `{north, south, east, west}`?

**Comment by eigengrau** [comment]
I agree with that constraint. FWIW I also don’t think any window that isn’t visible should be considered «to the X side of Y» for any Y.

Floating windows don’t really seem to mix well with a KISS approach to this, since for all intents and purposes north, south, etc. only make sense in regards to the border of the original window. When something is contained within the original window, it’s unclear what the origin of the direction is supposed to be (upper left corner, center, etc.?).

**Comment by aminb** [comment]
So if I understand correctly, that constraint would help with my issue if the two windows are of the exact *same size*, correct? If yes, any suggestions as to what to do if there's a *smaller* window behind the current large one?

**Comment by baskerville** [comment]
@aminb I reverted the aforementioned commit.

@eigengrau I'm using the [right-window](https://github.com/ntrrgc/right-window) algorithm. If by *hidden* you mean: *obscured by other windows*, then e2a959c is only a partial fix for this issue.

**Comment by aminb** [comment]
@baskerville Thanks. But right now if the triangles are equal I still can't do `super + {j,k}` to switch between them.. And I guess that's kinda the whole point with the monocle issue? It's better than the other commit, but still kinda inconvenient haha.

**Comment by baskerville** [comment]
@aminb Add `bspc config directional_focus_tightness low` to your *bspwmrc*.

**Comment by aminb** [comment]
@baskerville Perfect! Thanks.

**Comment by Alexander-Shukaev** [comment]
Can this one be closed?
